### PR TITLE
Show solana name service name as first letter of avatar

### PR DESCRIPTION
**Pull request by 0xavro**
avatar-sns

Show SNS letter as the first letter of the avatar

### CODE REVIEW

**Comment by fsher** [comment]
I see that we are loading the name tag twice: in threads list and when we open the thread. Do you have an idea what's that happening?

**Review by fsher — APPROVED** [review]
Well done! Approving in advance, left a couple of comments to discuss.

**Comment by 0xavro** [comment]
> I see that we are loading the name tag twice: in threads list and when we open the thread. Do you have an idea what's that happening?

not sure what's happening there, but it looks like it happens for the Twitter image as well. I can take a look at that, I think SWR should handle it.

File: packages/dialect-react-ui/components/Avatar/index.tsx [hunk 1]
```diff
@@ -55,6 +57,9 @@ export default function Avatar({ publicKey, size = 'regular' }: PropTypes) {
   const { avatar } = useTheme();
   const { program } = useApi();
 
+  const connection = program?.provider.connection;
+  const { data } = useSWR([connection, publicKey], fetchSolanaNameServiceName);
```
[the remark this answers is not in the document]
Comment (0xavro) [reply packages/dialect-react-ui/components/Avatar/index.tsx:61]: Yeah let's go with strings here instead

File: packages/dialect-react-ui/components/Avatar/index.tsx [hunk 2]
```diff
@@ -64,12 +69,16 @@ export default function Avatar({ publicKey, size = 'regular' }: PropTypes) {
...
       <div className={`${textStyleMap[size]}`}>
-        {program?.provider.connection ? (
+        {connection ? (
           <CardinalAvatar
             className="dt-h-full"
-            connection={program?.provider.connection}
+            connection={connection}
             address={publicKey}
-            placeholder={publicKey.toString().substr(0, 1)}
+            placeholder={
```
[the remark this answers is not in the document]
Comment (0xavro) [reply packages/dialect-react-ui/components/Avatar/index.tsx:77]: Yeah, I think we should allow the user to choose which component has precedence over another. I plan on refactoring this component to support that logic but that shouldn't block this PR. Let me know your thoughts @fsher

[the remark this answers is not in the document]
Comment (fsher) [reply packages/dialect-react-ui/components/Avatar/index.tsx:77]: Sounds good, please leave a TODO for this

**Pull request merged**